app/(nms)/inventory/gnbs_modals.tsx

```diff
@@ -0,0 +1,156 @@
+    closeFn: () => void
+}
+
+export const GnbModal: React.FC<GnbModalProps> = ({
+    title,
+    initialValues,
+    onSubmit,
+    closeFn,
+  }) => {
+    const auth = useAuth()
+    const [apiError, setApiError] = useState<string | null>(null);
+    const queryClient = useQueryClient()
+  
+    const formik = useFormik<GnbFormValues>({
+      initialValues,
+      validationSchema: GnbSchema,
+      onSubmit: async (values) => {
+        try {
+          onSubmit({ ...values, });
+          closeFn();
+          setTimeout(async () => { // Wait 100 ms before invalidating due to a race condition
+            await queryClient.invalidateQueries({ queryKey: [queryKeys.gnbs] });
+          }, 100);
+        } catch (error) {
+          if (is401UnauthorizedError(error)) {
+              auth.logout();
+          } else {
+            setApiError("An unexpected error occurred.");
+          }
+        }
+      },
+    });
+
+    return (
+      <Modal
+        title={ title }
+        close={ closeFn }
+        buttonRow={
+          <>
+          <Button
+            appearance="positive"
+            onClick={formik.submitForm}
+            disabled={!(formik.isValid && formik.dirty)}
+            loading={formik.isSubmitting}
+          >
+            Submit
+          </Button>
+          <Button onClick={closeFn}>Cancel</Button>
+          </>
+        }
+      >
+        {apiError && <ErrorNotification error={apiError} />}
+        <Form>
+          <Input
+            id="name"
+            label="Name"
+            type="text"
+            required
+            stacked
+            disabled
+            placeholder="default-gnb"
+            {...formik.getFieldProps("name")}
+            error={formik.touched.name ? formik.errors.name : null}
+          />
+          <Input
+            id="tac"
+            label="TAC"
+            help="Tracking Area Code"
+            type="number"
+            required
+            stacked
+            min={1}
+            max={16777215}
+            placeholder="1"
+            {...formik.getFieldProps("tac")}
+            error={formik.touched.tac ? formik.errors.tac : null}
+          />
+        </Form>
+      </Modal>
+    );
+  };
+  
+  type editGnbModalProps = {
+    gnb: GnbItem
+    closeFn: () => void
+  }
+  
+  export function EditGnbModal({ gnb, closeFn }: editGnbModalProps) {
+    const auth = useAuth()
+    const handleSubmit = async (values: GnbFormValues) => {
+      await editGnb({
+          name: values.name,
+          tac: values.tac,
+          token: auth.user ? auth.user.authToken : ""
+        });
+    };
+  
+    const initialValues: GnbFormValues = {
+      name: gnb?.["name"] || "",
+      tac: gnb?.["tac"],
+    };
+  
+    return (
+      <>
+        <GnbModal
+          title={"Edit gNB: " + `${gnb["name"]}`}
+          initialValues={initialValues}
+          onSubmit={handleSubmit}
+          closeFn={closeFn}
+        />
+      </>
+    );
+  }
```

Comment:
```suggestion
            >
            </GnbModal>
        </>
    );
}

```
We can close the model and add an EOL.

Reply:
The [self-closing tag](https://react-cn.github.io/react/tips/self-closing-tag.html) is valid.

Reply:
OK.